R/zchunk_LA1322.Fert_china.R

```diff
@@ -0,0 +1,140 @@
+#' original data system was \code{LA1322.Fert.R} (gcam-china level1).
+#' @details Calculate input-output intensity coefficients and input energy for province fertilizer production from province shares of national values.
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter mutate select
+#' @importFrom tidyr gather spread
+#' @author Yang Aug 2018
+module_gcam.china_LA1322.Fert <- function(command, ...) {
+  if(command == driver.DECLARE_INPUTS) {
+    return(c("L101.NBS_use_all_Mtce",
+             "L1322.Fert_Prod_MtN_R_F_Y",
+             "L1322.IO_R_Fert_F_Yh"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L1322.out_Mt_province_Fert_Yh",
+             "L1322.IO_GJkg_province_Fert_F_Yh",
+             "L1322.in_EJ_province_Fert_Yh"))
+  } else if(command == driver.MAKE) {
+
+    all_data <- list(...)[[1]]
+
+    # Load required inputs
+    L1322.Fert_Prod_MtN_R_F_Y <- get_data(all_data, "L1322.Fert_Prod_MtN_R_F_Y")
+    L1322.IO_R_Fert_F_Yh <- get_data(all_data, "L1322.IO_R_Fert_F_Yh")
+    # Proxy for downscaling fertilizer to provinces
+    # TODO: Find a better proxy
+    L101.NBS_use_all_Mtce <- get_data(all_data, "L101.NBS_use_all_Mtce")
+
+    # Silence package check
+    year <- GCAM_region_ID <- value <- org <- sum <- province <-
+      sector <- fuel <- EBProcess <- EBMaterial <- multiplier <- NULL
+
+    # ===================================================
+    # Get province fertilizer production, input-output intensity coefficients, and
+    # input energy for fertilizer production from national values.
+
+    # Assigning national fertilizer production to provinces
+    L101.NBS_use_all_Mtce %>%
+      filter(EBProcess == "Agriculture" & EBMaterial == "Coal Raw") %>%
+      # First zero out NAs in years where some values are NA but not all
+      group_by(province, year) %>%
+      mutate(value = replace(value, is.na(value) & sum(value, na.rm = T) != 0, 0)) %>%
+      ungroup ->
+      L1322.in_Mtce_province_Fert
+
+    L1322.in_Mtce_province_Fert %>%
+      group_by(province, sector, fuel) %>%
+      # use approx_fun rule = 2 to fill out data in years where the entire province is NA
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup %>%
+      # When there is one data point, approx_fun replaced it with NA, but what we needed is to replace all NAs with that data.
+      left_join(L1322.in_Mtce_province_Fert %>% rename(org = value),
+                by = c("province", "fuel", "sector", "year", "EBProcess", "EBMaterial")) %>%
+      group_by(sector, fuel, province) %>%
+      mutate(sum = sum(org, na.rm = T)) %>%
+      mutate(value = replace(value, is.na(value) & sum != 0, sum)) %>%
+      select(-org, -sum) %>%
+      ungroup ->
+      L1322.in_Mtce_province_Fert
+
+    # Convert to province precentages
+    L1322.in_Mtce_province_Fert %>%
```

Comment:
This section is not needed, can be done as:
`L1322.in_Mtce_province_Fert %>%`
    `group_by(sector, fuel, year) %>%`
    `mutate(multiplier = value / sum(value)) %>%`